src/backend/cuda/convolveNN_base.cpp

```diff
@@ -0,0 +1,190 @@
+/*******************************************************
+ * Copyright (c) 2014, ArrayFire
+ * All rights reserved.
+ *
+ * This file is distributed under 3-clause BSD license.
+ * The complete license agreement can be obtained at:
+ * http://arrayfire.com/licenses/BSD-3-Clause
+ ********************************************************/
+
+#include <Array.hpp>
+#include <blas.hpp>
+#include <cast.hpp>
+#include <common/half.hpp>
+#include <common/indexing_helpers.hpp>
+#include <common/unique_handle.hpp>
+#include <convolve.hpp>
```

Comment:
Header format:

```
<single header inclusion that has implementation definitions>
<empty newline>
<our internal backend headers>
<empty newline>
<system headers>
```
The rest of formatting will be taken care of by clang-format.